jubilant/_juju.py

```diff
@@ -53,40 +60,43 @@ def add_model(
                 args.extend(['--config', f'{k}={v}'])
 
         self.cli(*args)
+        self.model = model
+
+    def switch(self, model: str):
+        """TODO."""
+        self.cli('switch', model)
+        self.model = model
 
     def destroy_model(
         self,
-        model_name: str,
+        model: str,
         *,
         force=False,
     ):
         """TODO."""
-        args = ['destroy-model', model_name, '--no-prompt']
+        args = ['destroy-model', model, '--no-prompt']
         if force:
             args.append('--force')
         self.cli(*args)
+        self.model = None
 
     def deploy(
         self,
-        charm_name: str,
-        application_name: str | None = None,
+        charm: str,
```

Comment:
That's part of the reason I changed it to just "charm" instead of "charm_name". However, the typing suggestion is a good one -- I'll update. I guess the type would be `str | os.PathLike`.